File: problems/37-bitmask-dp/003-assignment-problem/java/Solution.java
import java.util.*;
public class Solution {
 public int minCost(int[][] cost){int n=cost.length,N=1<<n; int[] dp=new int[N]; Arrays.fill(dp,1_000_000_000); dp[0]=0; for(int mask=0;mask<N;mask++){int worker=Integer.bitCount(mask); if(worker==n)continue; for(int job=0;job<n;job++)if((mask&(1<<job))==0) dp[mask|1<<job]=Math.min(dp[mask|1<<job],dp[mask]+cost[worker][job]);} return dp[N-1];}
}

File: problems/37-bitmask-dp/003-assignment-problem/python/solution.py
class Solution:
    def min_cost(self,cost):
        n=len(cost); dp=[10**18]*(1<<n); dp[0]=0
        for mask in range(1<<n):
            worker=mask.bit_count()
            if worker==n: continue
            for job in range(n):
                if not mask>>job&1:
                    nxt=mask|1<<job
                    dp[nxt]=min(dp[nxt],dp[mask]+cost[worker][job])
        return dp[-1]
